Automattic/sensei
# Add color fallback to course categories block

Fixes #5533

### Changes proposed in this Pull Request
* Add a fallback when is not possible to load the theme colors.

### Testing instructions
* Create a page
* Add the Course List
* Enable the DIVI or Astra themes
* Check if the course categories are using the fallback color
* Enable Twenty Twenty
* Check if the course categories are using the colors from the theme.


#### Extra Notes
This approach is the same that we are using on other blocks, I just noticed that in a good amount of themes the withDefaultColors HOS frequently fall back to `dark-gray` and `white`, like DIV.

I tried a lot of approaches to find a more reliable way, but I found cons
The best that I tried was to get the highest contrast colors from the theme using this [lib](https://github.com/omgovich/colord), but the problem is that we need to store the selected colors in order to render it.
Storing the colors blocks the user of change the site theme and automatically see the changes on all pages.
